mlflow/pyspark/ml/__init__.py

```diff
@@ -29,22 +32,34 @@ def _read_log_model_allowlist_from_file(allowlist_file):
     return allowlist
 
 
-def _read_log_model_allowlist(spark_session):
+def _read_log_model_allowlist():
     """
     Reads the module allowlist and returns it as a set.
     """
+    from mlflow.utils._spark_utils import _get_active_spark_session
+
+    builtin_allowlist_file = resource_filename(__name__, "log_model_allowlist.txt")
+    spark_session = _get_active_spark_session()
+    if not spark_session:
+        _logger.warning(
+            "There's no spark session, cannot read config "
+            "'spark.mlflow.pysparkml.autolog.logModelAllowlistFile', "
+            "using default log model allowlist."
```

Comment:
Nit: I might lean towards us logging an info-level message rather than a warning here, since most users who fall into this case are probably not trying to specify a custom allowlist (but instead just called `autolog` before instantiating their SparkSession) and so shouldn't need to take any action. Maybe something like:

"No SparkSession detected. Autologging will log pyspark.ml models contained in the default allowlist. To specify a custom allowlist, initialize a SparkSession prior to calling mlflow.pyspark.ml.autolog() and specify the path to your allowlist file via the spark.mlflow.pysparkml.autolog.logModelAllowlistFile conf."

Also, note that we could avoid needing to show this message by reading the allowlist when `fit` is first called and caching the result (at which time I believe we're guaranteed to have a SparkSession accessible)